### PR TITLE
Added new query for EC2 network ACL overlapping ports

### LINKED ISSUE #1016: Add EC2 network ACL overlapping ports query for CloudFormation
### Platform
CloudFormation

### Provider
CloudFormation

### Description
"NetworkACL Entries are reusing or overlapping ports which may create ineffective rules

### PR DESCRIPTION
**Pull request by soniantao**
Closes #1016
Closes #1017

### CODE REVIEW

File: assets/queries/cloudFormation/ec2_network_Acl_overlap_ports/query.rego [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  prop1.Type == "AWS::EC2::NetworkAclEntry"
+  prop2.Type == "AWS::EC2::NetworkAclEntry"
+  
+  from1 := prop1.Properties.PortRange.From
+  to1 := prop1.Properties.PortRange.To
+  
+  from2 := prop2.Properties.PortRange.From
+  to2 := prop2.Properties.PortRange.To
+	
+  overlap([from1, to1], [from2, to2])
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/ec2_network_Acl_overlap_ports/query.rego:19]: we can simplify this check like this:

```
  range1 := numbers.range(from1, to1)
  range2 := numbers.range(from2, to2)
  check_overlap(range1, range2)
```
and then create a function called check overlap:
```
check_overlap(range1, range2) {
    some j,k
    range1[j] = range2[k]
}
```

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
Requesting changes

File: assets/queries/cloudFormation/ec2_network_Acl_overlap_ports/query.rego (hunk 1 above)
Comment (soniantao) [reply assets/queries/cloudFormation/ec2_network_Acl_overlap_ports/query.rego:19]: amazing comment. In fact, I did not remember to use this _numbers.range_ built-in function. This really works. I'm going to change it

File: assets/queries/cloudFormation/ec2_network_Acl_overlap_ports/metadata.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "ec2_network_ACL_overlapping_ports",
+  "queryName": "EC2 network ACL overlapping ports",
+  "severity": "HIGH",
+  "category": null,
```
[the remark this answers is not in the document]
Comment (soniantao) [reply assets/queries/cloudFormation/ec2_network_Acl_overlap_ports/metadata.json:5]: correct

**Comment by rogeriopeixotocx** [comment]
LGTM ✅

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Pull request merged**